src/web/page/templates.rs

```diff
@@ -42,19 +42,31 @@ impl TemplateData {
     }
 
     pub fn start_template_reloading() {
-        use std::{sync::Arc, thread, time::Duration};
-
-        thread::spawn(|| loop {
-            match load_templates() {
-                Ok(templates) => {
-                    log::info!("Reloaded templates");
-                    TEMPLATE_DATA.templates.swap(Arc::new(templates));
-                    thread::sleep(Duration::from_secs(10));
-                }
+        use notify::{watcher, RecursiveMode, Watcher};
+        use std::{
+            sync::{mpsc::channel, Arc},
+            thread,
+            time::Duration,
+        };
+
+        let (tx, rx) = channel();
+        let mut watcher = watcher(tx, Duration::from_secs(2)).unwrap();
```

Comment:
I had to look in the documentation to see what the duration was, a comment would be nice.